Modules/errnomodule.c

```
@@ -66,46 +66,17 @@ _inscode(PyObject *d, PyObject *de, const char *name, int code)
     Py_XDECREF(v);
 }
 
-PyDoc_STRVAR(errno__doc__,
-"This module makes available standard errno system symbols.\n\
-\n\
-The value of each symbol is the corresponding integer value,\n\
-e.g., on most systems, errno.ENOENT equals the integer 2.\n\
-\n\
-The dictionary errno.errorcode maps numeric codes to symbol names,\n\
-e.g., errno.errorcode[2] could be the string 'ENOENT'.\n\
-\n\
-Symbols that are not relevant to the underlying system are not defined.\n\
-\n\
-To map error codes to error messages, use the function os.strerror(),\n\
-e.g. os.strerror(2) could return 'No such file or directory'.");
-
-static struct PyModuleDef errnomodule = {
-    PyModuleDef_HEAD_INIT,
-    "errno",
-    errno__doc__,
-    -1,
-    errno_methods,
-    NULL,
-    NULL,
-    NULL,
-    NULL
-};
-
-PyMODINIT_FUNC
-PyInit_errno(void)
+static int
+errno_exec(PyObject *module)
 {
-    PyObject *m, *d, *de;
-    m = PyModule_Create(&errnomodule);
-    if (m == NULL)
-        return NULL;
-    d = PyModule_GetDict(m);
-    de = PyDict_New();
-    if (!d || !de || PyDict_SetItemString(d, "errorcode", de) < 0)
-        return NULL;
+    PyObject *module_dict = PyModule_GetDict(module);
+    PyObject *error_dict = PyDict_New();
+    if (!module_dict || !error_dict || PyDict_SetItemString(module_dict, "errorcode", error_dict) < 0) {
+        return -1;
+    }
 
 /* Macro so I don't have to edit each and every line below... */
-#define inscode(d, ds, de, name, code, comment) _inscode(d, de, name, code)
+#define inscode(module_dict, error_dict, name, code, comment) _inscode(module_dict, error_dict, name, code)
```

While we are here, you may take the opportunity to enhance error handling: add an result value to _inscode() and exit early errno_exec() at the first error.

Oh. It seems like right now, all errors are silently ignored! I expect that the caller will fail if the function leaks an exception but don't return NULL.

Why not make the first two arguments implicit? To write like:

```
inscode("ENODEV", ENODEV, "No such device");
```

"ins" stands for "insert". That's non obvious, especially that the code works on dict. On a dict, we "set" items, we don't "insert" them.

Maybe we can also take this opportunity to use a better name, like "add_errcode".